### PR TITLE
ci(workflow): add cache to workflows using actions/setup-node

### PR DESCRIPTION
**Pull request by oscard0m**
## Description

Add `cache` to workflows using `actions/setup-node`

## Context

`setup-node` GitHub Action just released a new option to add cache to steps using it.

You can find the details here: https://github.blog/changelog/2021-07-02-github-actions-setup-node-now-supports-dependency-caching/

---

🤖 This PR has been generated automatically by [this octoherd script](https://github.com/oscard0m/octoherd-script-add-cache-to-node-github-action), feel free to run it in your GitHub user/org repositories! 💪🏾

### CODE REVIEW

**Review by marcoow — APPROVED** [review]
thanks 🙌

**Comment by marcoow** [comment]
very strange though – one of the jobs keeps timing out…

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -4,10 +4,10 @@ on:
   push:
     branches:
       - master
-      - 'v*'
+      - "v*"
```
Comment (oscard0m) [inline .github/workflows/ci.yml:7]:
```suggestion
      - 'v*'
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -4,10 +4,10 @@ on:
...
   pull_request: {}
   schedule:
-    - cron:  '0 3 * * *' # daily, at 3am
+    - cron: "0 3 * * *" # daily, at 3am
```
Comment (oscard0m) [inline .github/workflows/ci.yml:10]:
```suggestion
    - cron:  '0 3 * * *' # daily, at 3am
```

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -3,7 +3,7 @@ name: Release
 on:
   push:
     tags:
-      - '*'
+      - "*"
```
Comment (oscard0m) [inline .github/workflows/release.yml:6]:
```suggestion
      - '*'
```

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -12,10 +12,11 @@ jobs:
 
     steps:
       - uses: actions/checkout@v1
-      - uses: actions/setup-node@v1
+      - uses: actions/setup-node@v2
         with:
           node-version: 12.x
-          registry-url: 'https://registry.npmjs.org'
+          registry-url: "https://registry.npmjs.org"
```
Comment (oscard0m) [inline .github/workflows/release.yml:18]:
```suggestion
          registry-url: 'https://registry.npmjs.org'
```

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -12,10 +12,11 @@ jobs:
 
     steps:
       - uses: actions/checkout@v1
-      - uses: actions/setup-node@v1
+      - uses: actions/setup-node@v2
         with:
           node-version: 12.x
           registry-url: 'https://registry.npmjs.org'
+          cache: npm
```
Comment (oscard0m) [inline .github/workflows/release.yml:19]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -16,9 +16,10 @@ jobs:
 
     steps:
       - uses: actions/checkout@v1
-      - uses: actions/setup-node@v1
+      - uses: actions/setup-node@v2
         with:
           node-version: 10.x
+          cache: npm
```
Comment (oscard0m) [inline .github/workflows/ci.yml:22]:
```suggestion
          cache: yarn
```

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -45,9 +46,10 @@ jobs:
...
```
Comment (oscard0m) [inline .github/workflows/ci.yml:52]:
```suggestion
          cache: yarn
```

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -61,9 +63,10 @@ jobs:
...
```
Comment (oscard0m) [inline .github/workflows/ci.yml:69]:
```suggestion
          cache: yarn
```

**Comment by oscard0m** [comment]
> very strange though – one of the jobs keeps timing out…

Very strange... Checking the logs there is a strange missing test file or something like that.

---

I think the job should be manually killed: https://github.com/simplabs/ember-cookies/runs/2987332114?check_suite_focus=true

**Review by marcoow — APPROVED** [review]

**Pull request merged**